### PR TITLE
Challenges with boolean validated

### LINKED ISSUE #43: Send boolean if validated or not
- [x] nineChallengesBool

- [x] getChallengeByCategoryBool

### PR DESCRIPTION
**Pull request by arobaselisa**
getNineChallengesBool + getChallengesByCatBool
Closes #43

### CODE REVIEW

File: back/src/main/java/coronhackathon/backend/controller/mainController.java [hunk 1]
```diff
@@ -0,0 +1,669 @@
+package coronhackathon.backend.controller;
+
```
Comment (PaulJuillard) [inline back/src/main/java/coronhackathon/backend/controller/mainController.java:2]: This file should not exist anymore

File: back/src/main/java/coronhackathon/backend/repository/CompletedRepository.java [hunk 2]
```diff
@@ -9,6 +9,7 @@
 import java.util.Optional;
 
 public interface CompletedRepository extends JpaRepository<HasCompleted, Long> {
+    //instead of Long it is possible it is HasCompletedId
```
Comment (PaulJuillard) [inline back/src/main/java/coronhackathon/backend/repository/CompletedRepository.java:12]: I don't understand what this means

Comment (Mathiponds) [reply back/src/main/java/coronhackathon/backend/repository/CompletedRepository.java:12]: I'm not sure but it is possible that what we wrote is wrong and insted of 
`JpaRepository<HasCompleted, Long>` we should write `JpaRepository<HasCompleted, HasCompletedId>`

File: back/src/main/java/coronhackathon/backend/service/ChallengeService.java [hunk 3]
```diff
@@ -58,4 +68,29 @@ public Long numberOfChallengesByCategoryName(String name) {
         Collections.shuffle(all);
         return all.subList(0, Math.min(9, all.size()));
     }
+
+    public List<Boolean> getNineBoolean(String username, List<Challenge> lc){
```
Comment (Mathiponds) [inline back/src/main/java/coronhackathon/backend/service/ChallengeService.java:72]: Can you add the doc for both functions ?

File: back/src/main/java/coronhackathon/backend/service/ChallengeService.java [hunk 4]
```diff
@@ -58,4 +68,29 @@ public Long numberOfChallengesByCategoryName(String name) {
...
+        if (!ou.isPresent())
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "user with name : " + username + " not found");
+        List<Challenge> cc = completedService.getCompletedChallenges(username);
+        for(Challenge c : lc) {
+            lb.add(cc.contains(c));
+        }
+        return lb;
+    }
+
+    public List<Boolean> getChallengeBool(String username, long categoryId){
```
Comment (Mathiponds) [inline back/src/main/java/coronhackathon/backend/service/ChallengeService.java:84]: Are we sure the list will have the same order as the order of the challenge in the list ?

Comment (arobaselisa) [reply back/src/main/java/coronhackathon/backend/service/ChallengeService.java:84]: Yes, the list is ordered so it should always work. However, one problem might arise : if a challenge is added/deleted between the two times we go through the list. I assumed this wouldn't happen, but maybe I need to be more careful of that ?

File: back/src/main/java/coronhackathon/backend/service/ChallengeService.java (hunk 3 above)
Comment (arobaselisa) [reply back/src/main/java/coronhackathon/backend/service/ChallengeService.java:72]: I added them, I pushed the file, I don't know if it worked ?

File: back/src/main/java/coronhackathon/backend/service/ChallengeService.java (hunk 4 above)
Comment (Mathiponds) [reply back/src/main/java/coronhackathon/backend/service/ChallengeService.java:85]: Let's say it would work

**Pull request merged**